src/app/topic-creation/topic-creation/topic-creation.component.ts

```diff
@@ -11,79 +10,51 @@ import {ActivatedRoute} from '@angular/router';
 })
 export class TopicCreationComponent implements OnInit {
 
-  loading = true;
   formGroup: FormGroup;
-  advancedConfigForm: XmppDataForm;
+  error: string;
   isNewCollection: boolean;
 
-  private xmppDataForm: XmppDataForm;
-  private readonly specificFormFields = {
-    'pubsub#title': new FormControl(null, Validators.required)
-  };
-
   constructor(private creationService: TopicCreationService,
               private navigationService: NavigationService,
               private route: ActivatedRoute) {
   }
 
   ngOnInit(): void {
     this.isNewCollection = this.route.snapshot.data.type === 'collection';
-    this.creationService.loadForm().then((form: XmppDataForm) => {
-
-      this.specificFormFields['pubsub#node_type'] = new FormControl(this.route.snapshot.data.type);
-      this.specificFormFields['pubsub#children'] = new FormControl(null);
-      this.specificFormFields['pubsub#collection'] = new FormControl(null);
-
-      this.formGroup = new FormGroup(this.specificFormFields);
-      this.advancedConfigForm = this.filterForm(form);
-
-      this.xmppDataForm = form;
-      this.loading = false;
+    this.formGroup = new FormGroup({
+      'nodeID': new FormControl(null)
     });
   }
 
-  submit(): void {
-    if (!this.formGroup.valid) {
-      return;
-    }
-    const fields = [];
-    this.xmppDataForm.fields.forEach((field: XmppDataFormField) => {
-        const newValue = this.formGroup.get(field.variable).value;
-        if (newValue === field.value) {
-          return;
-        }
-        fields.push(new XmppDataFormField(
-          field.type,
-          field.variable,
-          newValue,
-          field.label,
-          field.options
-        ));
+  submit() {
+    this.formGroup.disable();
+    this.creationService.createTopic(this.formGroup.get('nodeID').value)
+      .then((topicIdentifier) =>
+        this.navigationService.goToTopic(topicIdentifier)
+      ).catch((error) => {
+      this.formGroup.enable();
+      switch (error.condition) {
+        case TopicCreationErrors.FeatureNotImplemented:
+          this.error = 'Service does not support node creation';
+          break;
+        case TopicCreationErrors.RegistrationRequired:
+          this.error = 'Service requires registration';
+          break;
+        case TopicCreationErrors.Forbidden:
+          this.error = 'Requesting entity is prohibited from creating nodes';
+          break;
+        case TopicCreationErrors.Conflict:
+          this.error = 'A topic with the given identifier does already exist';
+          break;
+        case TopicCreationErrors.NodeIdRequired:
+          this.error = 'Service does not support instant nodes';
+          break;
+        default:
+          console.log(error);
```

Comment:
Is this a debug artefact? If not, we should mention in the log message why we logged it (Console outputs like `Object{condition: '?!?!', type: 'unknown'}` are not helpful if you don't know were to look 😉)

Reply:
This was actually intended - in case of an unknown error it's nice to check the console for further debug information...but i understand that it can be confusing in the log output...should I delete it then?

Reply:
No, let's keep it there for the moment. I just saw it in the test output; in the browser, there is a reference to where it was logged, so that's what I originally was going for 😊